orc/src/main/java/org/apache/iceberg/orc/ORC.java

```diff
@@ -143,12 +152,73 @@ public WriteBuilder metricsConfig(MetricsConfig newMetricsConfig) {
       return this;
     }
 
+    // supposed to always be a private method used strictly by data and delete write builders
+    private WriteBuilder createContextFunc(Function<Configuration, Context> newCreateContextFunc) {
+      this.createContextFunc = newCreateContextFunc;
+      return this;
+    }
+
     public <D> FileAppender<D> build() {
       Preconditions.checkNotNull(schema, "Schema is required");
+
+      // Map Iceberg properties to pass down to the ORC writer
+      Context context = createContextFunc.apply(conf);
+      long stripeSize = context.stripeSize();
+      long blockSize = context.blockSize();
+
+      conf.setLong(OrcConf.STRIPE_SIZE.getAttribute(), stripeSize);
+      conf.setLong(OrcConf.BLOCK_SIZE.getAttribute(), blockSize);
+
       return new OrcFileAppender<>(schema,
           this.file, createWriterFunc, conf, metadata,
           conf.getInt(VECTOR_ROW_BATCH_SIZE, VectorizedRowBatch.DEFAULT_SIZE), metricsConfig);
     }
+
+    private static class Context {
+      private final long stripeSize;
+      private final long blockSize;
+
+      public long stripeSize() {
+        return stripeSize;
+      }
+
+      public long blockSize() {
+        return blockSize;
+      }
+
+      private Context(long stripeSize, long blockSize) {
+        this.stripeSize = stripeSize;
+        this.blockSize = blockSize;
+      }
+
+      static Context dataContext(Configuration conf) {
+        String stripeSizeStr =
+            conf.get(TableProperties.ORC_STRIPE_SIZE_BYTES, conf.get(OrcConf.STRIPE_SIZE.getAttribute()));
```

Comment:
What's the difference between the `getAttribute` and `getHiveConfName` ?  Should fallback to read the hive conf key if the attribute is not present in this hadoop configuraiton ?

Reply:
Whether the following logic should be used:
1. First try to get `TableProperties.ORC_STRIPE_SIZE_BYTES`
2. If null, try `OrcConf.STRIPE_SIZE.getAttribute()`.
3. If `getAttribute` is null, try `getHiveConfName`.

Reply:
I mean if people just set their hiveConfName but don't set the key attribution name, then should we also parse the hiveConfName , right ?

Reply:
I think the correct approach to parse the config keys is: 

```java
static Context dataContext(Configuration conf) {
        long stripeSize = TableProperties.ORC_STRIPE_SIZE_BYTES_DEFAULT;
        stripeSize = conf.getLong(OrcConf.STRIPE_SIZE.getAttribute(), stripeSize);
        stripeSize = conf.getLong(OrcConf.STRIPE_SIZE.getHiveConfName(), stripeSize);
        stripeSize = conf.getLong(TableProperties.ORC_STRIPE_SIZE_BYTES, stripeSize);

        long blockSize = TableProperties.ORC_BLOCK_SIZE_BYTES_DEFAULT;
        blockSize = conf.getLong(OrcConf.BLOCK_SIZE.getAttribute(), blockSize);
        blockSize = conf.getLong(OrcConf.BLOCK_SIZE.getHiveConfName(), blockSize);
        blockSize = conf.getLong(TableProperties.ORC_BLOCK_SIZE_BYTES, blockSize);

        return new Context(stripeSize, blockSize);
      }
```

Reply:
Got it. it is much more graceful.